privilege/privileges/privileges.go

```diff
@@ -160,12 +160,20 @@ func (p *UserPrivileges) ConnectionVerification(user, host string, authenticatio
 }
 
 // DBIsVisible implements the Manager interface.
-func (p *UserPrivileges) DBIsVisible(db string) bool {
+func (p *UserPrivileges) DBIsVisible(activeRoles []*auth.RoleIdentity, db string) bool {
 	if SkipWithGrant {
 		return true
 	}
 	mysqlPriv := p.Handle.Get()
-	return mysqlPriv.DBIsVisible(p.user, p.host, db)
+	if mysqlPriv.DBIsVisible(p.user, p.host, db) {
```

Comment:
If `r_1` has privilege, `r_2` doesn't have, and has relationship like `r_1 -> r_2 -> user`, user should have privilege to visit db.

Reply:
Do you mean user has role r_2
r_2 has role r_1 ?

I think `activeRoles` contains both r_1 and r_2 ? @imtbkcat

Reply:
`ctx.ActiveRoles` just contain r_1, you could use `MySQLPrivileges.FindAllRole` to get r_1 and r_2.

Reply:
Done.
PTAL @imtbkcat